maven/lib/dependabot/maven/update_checker/version_finder.rb

```diff
@@ -154,25 +155,28 @@ def released?(version)
         end
 
         def dependency_metadata(repository_details)
-          @dependency_metadata ||= {}
-          @dependency_metadata[repository_details.hash] ||=
```

Comment:
This memoization pattern works, but I found it hard to read so this is an opinionated change just to improve the readability and make this a little easier to maintain in the event that the `fetch_dependency_metadata` code was ever to return a falsy value rather than a blank string.